docs/Manifests.md

```diff
@@ -1,6 +1,190 @@
+
+### Components top-level element
+
+An AppManifest must declare a `components` root element as a list of objects containing configuration for each component in your application. At least one component object must be defined.
+
+```yaml
+components:
+  - name: webserver
+    ...
+  - name: api
+    ...
+```
+
+<!-- <details> -->
+
+#### name
+
+Defines the name of the component. Each component `name` must be unique. **Required**.
+
+#### type
+
+Specifies the type of the component; must be one of `service`, `task`, or `static`. **Required**.
+
+#### image
+
+Sets the base image from which the component is built. **Required**.
+
+#### port
+
+The container port to expose.
+
+#### root
+
+Establishes the docker build context, setting the host working directory for `copy` and `run` build step commands.
+
+#### build
+
+Defines a series of `steps` to be executed in creating the component image. Steps are defined as an array of mappings, where the mapping key is the step instruction and mapping value are arguments for that instruction.
+
+Steps:
+
+##### directory
+
+Sets the working directory of the image for build step commands that follow.
+
+##### copy
+
+Copies files from the host to the image. The copy step can take both string and array types as an argument. Strings can use the wildcard symbol '\*' to pattern match files. Arrays must have at least one element, where elements represent individual files to be copied from host to image. The copy step object can be used in conjunction with the optional `destination` key to specify the image directory in which files are to be copied.
+
+```yaml
+...
+    build:
+      steps:
+        - directory: /express
+        - copy: package*.json
+          destination: ./
+        - copy: [index.js, dynamodb.js, s3.js]
+          destination: ./
+    ...
+```
+
+##### run
+
+Executes a command in the current working directory. If no directory step precedes the run step, the command will execute in the base image's working directory.
+
+##### image
+
+Specifies an image to be used for multi-stage builds. The optional `stage` key can be used to name the build stage. The build stage should be a `string` value and cannot equal `main`, which is the name of the primary build stage.
+
+```yaml
+    build:
+      steps:
+      ...
+        - image: node:10
+          stage: server-build
+      ...
+```
+
+#### runtime
+
+`runtime` defines properties associated with a component instance. `runtime` properties affect build [Deployments](/docs/Deployments.md), specifying container execution commands, instance memory, component resources, and component variables.
+
+##### resources
+
+Resources that a component will use must be defined in the `resources` property of a component's runtime section. Resources are an array of strings, where the string value corresponds to the name of the resource as defined in the AppManifest resources (provide link) section.
```

Comment:
Might be good to not that only resource names included in a Component's resource array will be accessible to a Component.